libs/vkd3d/swapchain.c

```diff
@@ -1577,6 +1647,31 @@ static void dxgi_vk_swap_chain_recreate_swapchain_in_present_task(struct dxgi_vk
 
     INFO("Got %u swapchain images.\n", chain->present.backbuffer_count);
 
+    /* If low latency is supported restore the current low latency state now */
+    if (chain->queue->device->vk_info.NV_low_latency2)
+    {
+        struct low_latency_state* low_latency_state = chain->present.low_latency_update_requested ?
+            &chain->present.requested_low_latency_state : &chain->present.low_latency_state;
+
+        memset(&swapchain_latency_sleep_mode_info, 0, sizeof(swapchain_latency_sleep_mode_info));
+        swapchain_latency_sleep_mode_info.sType = VK_STRUCTURE_TYPE_LATENCY_SLEEP_MODE_INFO_NV;
+        swapchain_latency_sleep_mode_info.pNext = NULL;
+
+        swapchain_latency_sleep_mode_info.lowLatencyMode = low_latency_state->mode;
+        swapchain_latency_sleep_mode_info.lowLatencyBoost = low_latency_state->boost;
+        swapchain_latency_sleep_mode_info.minimumIntervalUs = low_latency_state->minimum_interval_us;
+
+        VK_CALL(vkSetLatencySleepModeNV(chain->queue->device->vk_device, chain->present.vk_swapchain, &swapchain_latency_sleep_mode_info));
+
+        if (chain->present.low_latency_update_requested)
+        {
+            memcpy(&chain->present.low_latency_state, &chain->present.requested_low_latency_state, sizeof(struct low_latency_state));
+            chain->present.low_latency_update_requested = false;
+        }
+
+        chain->present.using_low_latency_present_mode = dxgi_vk_swap_chain_check_low_latency_present_mode_support(chain, present_mode);
```

Comment:
With EXT_swapchain_maintenance1, this is naive since the present mode can change at will. Based on Ishi's comments, it seems like latency present mode support is merely a hint, not an actual requirement. Maybe this check can just be ignored?

Reply:
I did have an override for the EXT_swapchain_maintenance1 case but it isn't needed regardless. I have removed this check like you and ishitatsuyuki have suggested.